src/simsopt/geo/surface.py

```diff
@@ -425,7 +411,8 @@ def bisection(phia, a, phic, c):
                 err = np.max(np.abs(a - c))
             b = (a + c) / 2.
             return b
-
+        
+        phi = put_angle_between_left_and_right(phi, cyl_phi_left, cyl_phi_right)
         # bisect cyl_phi to compute the cross section
         sol = bisection(cyl_phi_left, varphi_left, cyl_phi_right, varphi_right)
```

Comment:
just trying to understand here: it looks like, for each `theta`, we are doing a bisection over `varphi`, so that `varphi2phi(varphi) = phi`. Why do we need to know `cyl_phi_right` and `cyl_phi_left`?